### ISSUE TITLE
Support for 4.4 Mongodb versions

### ISSUE THREAD

**Issue by schamane** [mongodb/mongodb-kubernetes-operator#161]
I tried to deploy cluster on kind with version set to 4.4.0 and it stuck on agent checking health.
It would be cool to support newest 4.4 mongo too.

**Comment by bznein** [comment]
Hi @schamane thanks for the report.

Can you please provide more context for us to reproduce/understand the issue?
Can you show us the specification of the deployment you are trying to create and the logs from the operator?

Thanks

**Comment by LouisPlisso** [comment]
@schamane this is because we didn't upgrade the automation agent since the release of 4.2: once this is done it should work directly

**Comment by LouisPlisso** [comment]
@chatton can you please take a look?

**Comment by chatton** [comment]
Hi @schamane we'll be upgrading the automation agent to a version which supports mongodb 4.4. So 4.4 support will be available in the next operator release.

**Comment by rajavignesh** [comment]
Facing issue while upgrading from version 4.4.1 to 4.4.2.
Using the latest agent, as described in pod
"Successfully pulled image "quay.io/mongodb/mongodb-agent:10.19.0.6562-1"

Seeing "Readiness probe failed" log in the pod and "StatefulSet default/ea-mongodb-dba is not yet ready, retrying in 10 seconds" message in the object description.

When deploying 4.4.1 or 4.4.2 there is no issue. Not able to upgrade within 4.4.

Please share your input.

**Comment by chatton** [comment]
Hi @rajavignesh, 

The problem you're describing sounds a little different that the initial problem, would you be able to create a new issue for this?

**Comment by rajavignesh** [comment]
Hi @chatton 
Opened a new ticket (#278) for the upgrade.
Thank you.
